.github/workflows/integration_test.yml

```diff
@@ -0,0 +1,259 @@
+        run: |
+          set +e
+          count=0
+          until $(psql -h 127.0.0.1 -U postgres -p 5432 -l > /dev/null); do
+            if [[ "$count" == "25" ]]; then
+              exit 1
+            fi
+            (( count++ ))
+            printf '.'
+            sleep 5
+          done
+          set -e
+      - name: Install plugin specific MySQL DDL
+        if: matrix.docker-compose-file == 'docker-compose.ci.mysql.yml' && inputs.ddl_file != ''
+        run: |
+          cat ${{ inputs.ddl_file }} | mysql -h 127.0.0.1 -u root --password=root killbill
+      - name: Install plugin specific PostgreSQL DDL
+        if: matrix.docker-compose-file == 'docker-compose.ci.postgresql.yml' && inputs.ddl_file != ''
+        run: |
+          cat ${{ inputs.ddl_file }} | psql -h 127.0.0.1 -U postgres -p 5432 -d killbill
+      - name: Start Kill Bill
+        # Sometimes it gets stuck (if Kill Bill starts when the DB isn't ready?)
+        timeout-minutes: 4
+        run: |
+          docker start it_killbill_1
+          count=0
+          until $(curl --connect-timeout 10 --max-time 30 --output /dev/null --silent --fail http://${KB_ADDRESS}:${KB_PORT}/1.0/healthcheck); do
+            if [[ "$count" == "180" ]]; then
+              exit 64
+            fi
+            count=$(( count + 1 ))
+            sleep 1
+          done
+          curl --connect-timeout 10 --max-time 30 -v \
+               -X POST \
+               -u admin:password \
+               -H 'Content-Type: application/json' \
+               -H 'X-Killbill-CreatedBy: GitHub' \
+               -d '{"apiKey": "bob", "apiSecret": "lazar"}' \
+               "http://${KB_ADDRESS}:${KB_PORT}/1.0/kb/tenants"
+      - name: Verify plugin has started
+        run: |
+          sudo snap install yq
+          count=0
+          until [[ "$(curl -s -uadmin:password http://${KB_ADDRESS}:${KB_PORT}/1.0/kb/nodesInfo | jq -r '.[].pluginsInfo[] | select(.pluginName == "${{ inputs.plugin_name }}-plugin").state')" == "RUNNING" ]]; do
+            if [[ "$count" == "5" ]]; then
+              curl -uadmin:password http://${KB_ADDRESS}:${KB_PORT}/1.0/kb/nodesInfo | jq .
+              exit 64
+            fi
+            count=$(( count + 1 ))
+            sleep 1
+          done
+      - name: Checkout killbill-integration-tests
+        uses: actions/checkout@v2
+        with:
+          repository: killbill/killbill-integration-tests
+          ref: ${{ inputs.integration_tests_ref }}
+          path: killbill-integration-tests
+      - name: Run tests
+        env:
+          DB_ADAPTER: ${{ matrix.database-adapter }}
+          DB_USER: ${{ matrix.database-user }}
+          DB_PASSWORD: ${{ matrix.database-password }}
+          DB_PORT: ${{ matrix.database-port }}
+        run: |
+          cd killbill-integration-tests
+          bundle install --jobs=4 --retry=3
+          bundle exec rake ${{ inputs.integration_tests_goal }}
```

Comment:
Perhaps, we could have a rake task for each plugin, i.e use the `plugin_name` as the name of the rake task?

Reply:
We do :-) See https://github.com/killbill/killbill-stripe-plugin/blob/9d86e94a14ea725a784c725f11d2385cd38bfebe/.github/workflows/ci.yml#L19 for instance.